tasks/configure.yml

```diff
@@ -35,6 +35,44 @@
   tags:
     - rstudio-connect-configure-environment-file
 
+- name: Configure | discover Python versions
+  block:
+    - name: Configure | find directories with Python
+      ansible.builtin.find:
+        paths: "{{ python_main_directory }}"
+        recurse: false
+        file_type: directory
+      register: python_paths
+
+    - name: Configure | set discovered_python_executables
+      ansible.builtin.set_fact:
+        # [objects] -> ['/opt/python/3.10.6',...] ->  [['/opt/python/3.10.6','/bin/python3'],...] -> ['/opt/python/3.10.6/bin/python3',...]
+        discovered_python_executables: "{{ python_paths.files | map(attribute='path') | product([python_sufix_bin]) | map('join') }}"
+
+    # admin specified python_versions - get only matching executable paths
+    - name: Configure | set (filtered) executables
+      ansible.builtin.set_fact:
+        python_executables: "{{ python_executables | d([]) + [item] }}"
+      loop: "{{ discovered_python_executables }}"
+      # spliting item: '/opt/python/3.10.6/bin/python3' -> ['', '3.10.6/bin/python3'] ->
+      # -> '3.10.6/bin/python3' -> ['3.10.6', ''] -> '3.10.6'
+      when: python_versions | length != 0 and (item.split(python_main_directory)[1].split(python_sufix_bin)[0] in python_versions)
+
+    # admin not specified python_versions - get all executable paths
+    - name: Configure | set (unfiltered) executables
+      ansible.builtin.set_fact:
+        python_executables: "{{ discovered_python_executables }}"
+      when: python_versions | length == 0
+
+    - name: Configure | set rstudio_connect_python_executables
+      ansible.builtin.set_fact:
+        rstudio_connect_python_executables: "{{ python_executables }}"
+
+  # discover Python executables only if admin doesn't specify concrete paths in rstudio_connect_python_executables
+  when: rstudio_connect_enable_python and (rstudio_connect_python_executables is not defined or rstudio_connect_python_executables | length == 0)
+  tags:
+    - rstudio-connect-configure-discover-python-executables
```

Comment:
This step perform discovery/search of Python executables. It's only executed when admin doesn't provide `rstudio_connect_python_executables` ("static", concrete paths to Python).

Variable `python_versions` can be specified in [ansible-python-install](https://github.com/Appsilon/ansible-python-install#role-variables). This role may be used in conjunction with this role (ansible-rstudio-connect) - admin installs Pythons and then RStudio Connect.
When `python_versions` is specified then the role will filter found Python executables and add to configuration only matching versions.
If `python_versions` is empty list, then all discovered executables will be added to the configuration.

Reply:
At first I tried to implement this in the template - it worked, however it was unreadable for me ([awful template changes in different branch](https://github.com/Appsilon/ansible-rstudio-connect/commit/6e9ede0b6472d79644ffbaebeffee277212e2307#diff-dd5e9393b71623e167fcfaeb7055d6c619a68a7ca3194e18b7a72de364e5628d))

Reply:
Seems complicated.

Let's look at it this way. I'm using the `rstudio-connect-ansible` role to install RStudio Connect. I want python, so I specify 'python_versions = 3.9.1' and I assume it will be enough to install python, enable it in configuration and set correct executables. So by default, when `python_versions` is defined then the `rstudio_connect_enable_python` should be true, and Executables should be set to the ones created by our ansible role for python, this can be achieved by simply specifying this in macro and changing only the versions like this (without using the `rstudio_connect_python_executables` variable) :
```
Executable = /opt/python/{{ version }}/bin/python3
```

However, if someone wants to install python using their own ansible role, then they have option to set `rstudio_connect_python_executables` manually and this should be also configured in macro as a `else` case. If this variable is setup then enable_python should also be automatically true.

In other cases enable_python should be false.

This is how I would approach it. Just to make it simpler for future maintenance. What do you think?

Reply:
I think you're right - my solution is too complicated. I will try to implement it the way you described

Reply:
Implemented in [the configuration template](https://github.com/Appsilon/ansible-rstudio-connect/pull/24/files#diff-dd5e9393b71623e167fcfaeb7055d6c619a68a7ca3194e18b7a72de364e5628dR11-R23)